app/lib/foreman/http_proxy.rb

```diff
@@ -12,6 +16,21 @@ def http_proxy_except_list
       end
     end
 
+    def http_proxy_full_url
+      return unless Setting[:http_proxy].present?
+
+      if USERINFO.one? { |key| Setting[key].present? }
+        raise Foreman::SettingValueException, N_("Must specify a HTTP(S) proxy user with password")
+      end
+
+      return Setting[:http_proxy] if USERINFO.none? { |key| Setting[key].present? }
+
+      url = URI.parse(Setting[:http_proxy])
+      url.user = CGI.escape(Setting[:http_proxy_username])
+      url.password = CGI.escape(Setting[:http_proxy_password])
```

Comment:
Is it really needed to CGI escape this? I'd think that `URI` itself takes care of that in `.to_s`.